libs/nx-plugin/src/executors/code-gen/executor.ts

```diff
@@ -4,19 +4,30 @@ import {
 } from '@featureboard/api-authentication'
 import type { FeatureBoardAuth } from '@featureboard/code-generator'
 import { codeGenerator } from '@featureboard/code-generator'
-import type { Tree } from '@nx/devkit'
-import { joinPathFragments, readProjectConfiguration } from '@nx/devkit'
+import type { ExecutorContext, Tree } from '@nx/devkit'
+import { joinPathFragments } from '@nx/devkit'
+import { FsTree, flushChanges, printChanges } from 'nx/src/generators/tree'
 import { API_ENDPOINT, CLIENT_ID } from '../../shared/config'
-import { isDryRun } from '../../shared/is-dry-run'
-import { isVerbose } from '../../shared/is-verbose'
 import type { CodeGenExecutorSchema } from './schema'
+
 export async function codeGenExecutor(
-    tree: Tree,
-    { projectName, ...options }: CodeGenExecutorSchema,
+    options: CodeGenExecutorSchema,
+    context: ExecutorContext,
+    tree?: Tree,
 ) {
-    const project = readProjectConfiguration(tree, projectName)
-    const dryRun = isDryRun()
-    const verbose = isVerbose()
+    if (!context.projectName) {
+        throw new Error('No projectName')
```

Comment:
nit: this could be more wordy and friendly perhaps?

Reply:
Updated